rfqmsg/messages.go

```diff
@@ -32,7 +33,27 @@ func (id ID) Scid() SerialisedScid {
 	scidBytes := id[24:]
 
 	scidInteger := binary.BigEndian.Uint64(scidBytes)
-	return SerialisedScid(scidInteger)
+
+	// We need to make sure the block height is within the range of valid
+	// SCID block heights.
+	scid := lnwire.NewShortChanIDFromInt(scidInteger)
+
+	minBlock := uint32(aliasmgr.AliasStartBlockHeight)
+	maxBlock := uint32(aliasmgr.AliasEndBlockHeight)
+
+	// If we're within the valid range, we can return the SCID as is.
+	if aliasmgr.IsAlias(scid) {
+		return SerialisedScid(scid.ToUint64())
+	}
+
+	// Normalize the value relative to min, then wrap within blockRange, and
+	// finally shift back by min. Generated by ChatGPT and I couldn't find a
+	// better way to do this.
+	blockRange := maxBlock - minBlock + 1
+	scid.BlockHeight = ((scid.BlockHeight-minBlock)%blockRange+blockRange)%
+		blockRange + minBlock
+
+	return SerialisedScid(scid.ToUint64())
```

Comment:
I think this shifting will need to be described in the BLIP:
```
Given a valid `rfq_id`, the RFQ specific SCID `tap_rfq_scid` is defined by
taking the last `8` bytes of the `rfq_id` and interpreting them as a 64-bit
integer.
```

It would probably be simpler at this point to just generate a SCID independent of RFQ ID and pass it to our peer in the RFQ request message.

There might be an advantage in separating the RFQ message ID from the SCID. We might be able to re-use the same SCID across multiple quotes for example. So every payment would correspond to a single RFQ quote but then multiple payments can share the same SCID.

There may be an advantage there when we come up with a formulation that avoids quote re-use.

Reply:
I don't think separating the two values is a good idea at this point. I went with the approach to generate random IDs until they fall within the correct range.

So the bLIP could just say: "The `rfq_id` must be drawn in a way that the derived `tap_rfq_scid` falls within the correct range for an SCID alias (`16_000_000 <= block height < 16_250_000`).